src/controller/greenPower.ts

```diff
@@ -47,44 +47,71 @@ class GreenPower extends events.EventEmitter {
                 dataPayload.frame.Payload.srcID, dataPayload.frame.Payload.commandFrame.securityKey
             );
 
-            const payload = {
-                options: 0x00e548,
-                srcID: dataPayload.frame.Payload.srcID,
-                sinkGroupID: this.adapter.greenPowerGroup,
-                deviceID: dataPayload.frame.Payload.commandFrame.deviceID,
-                frameCounter: dataPayload.frame.Payload.commandFrame.outgoingCounter,
-                gpdKey: [...key],
-            };
-
-            const frame = Zcl.ZclFrame.create(
-                Zcl.FrameType.SPECIFIC, Zcl.Direction.SERVER_TO_CLIENT, true,
-                null, ZclTransactionSequenceNumber.next(), 'pairing', 33, payload
-            );
-
-            await this.adapter.sendZclFrameToAll(242, frame, 242);
+            if (dataPayload.wasBroadcast) {
+                const payload = {
+                    options: 0x00e548,
+                    srcID: dataPayload.frame.Payload.srcID,
+                    sinkGroupID: this.adapter.greenPowerGroup,
+                    deviceID: dataPayload.frame.Payload.commandFrame.deviceID,
+                    frameCounter: dataPayload.frame.Payload.commandFrame.outgoingCounter,
+                    gpdKey: [...key],
+                };
+
+                const frame = Zcl.ZclFrame.create(
+                    Zcl.FrameType.SPECIFIC, Zcl.Direction.SERVER_TO_CLIENT, false,
+                    null, ZclTransactionSequenceNumber.next(), 'pairing', 33, payload
+                );
+
+                await this.adapter.sendZclFrameToAll(242, frame, 242);
+            }
+            else {
+                const coordinator = await this.adapter.getCoordinator();
```

Comment:
Could you explain this if/else logic?

Reply:
When permit join below is used towards an individual proxy (i.e. network address isn't null) I specified that the commissioning should be encapsulated as a unicast (0x2b, rather than 0x0b).  This means that when the commissioning packet is received here in this mode, it can be simply checked if it was broadcast or not.  If its not broadcast (i.e. unicast) I configure the payload for the pairing packet to instruct the proxy *which forwarded the commisioning packet* to forward messages via unicast for the given GPD.

This lets a person commissioning to choose the proxy (or if repeated proxies) they would like to use for GPDF forwarding.  This existing behaviour (groupcasting on all proxies if a packet is received) works, but scales poorly on large networks.

Reply:
Thanks for the explanation!